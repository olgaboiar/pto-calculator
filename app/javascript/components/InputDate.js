import React from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

class InputDate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      startDate: this.props.startDate,
    },
    this.handleChange = this.handleChange.bind(this);
  }
  
  handleChange(date) {
    this.setState({
      startDate: date
    });
    this.props.onChange(date);
  }
  
  render() {
    return (
      <div className="form-group row center-align" data-testid="startDate">
        <label>Select Your Start Date: </label>
        <DatePicker
          selected={this.state.startDate}
          onChange={this.handleChange}
          name="startDate"
          value={this.state.startDate}
        />
      </div>
    );
  }
}

export default InputDate;
