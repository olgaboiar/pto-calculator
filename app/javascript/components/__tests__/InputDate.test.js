import { render, mount } from "enzyme";
import React from "react";
import moment from "moment";
import InputDate from "../InputDate";
import 'react-datepicker/dist/react-datepicker.css';

describe("InputDate.js", () => {
  it("renders correctly", () => {  
    const wrapper = render(
      <InputDate />,
    );
  
    expect(wrapper).toMatchSnapshot();
  });

  it("displays an initial value", () => {
    const initialDate = moment("2019-01-01").toDate();

    const wrapper = mount(
      <InputDate 
        startDate={initialDate}
        onChange={jest.fn()}
      />);
    const input = wrapper.find("input");

    expect(input.props().value).toEqual("01/01/2019");
  });
});
