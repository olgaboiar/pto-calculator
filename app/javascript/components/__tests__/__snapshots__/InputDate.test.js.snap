// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDate.js renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "form-group row center-align",
      "data-testid": "startDate",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Select Your Start Date: ",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "label",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "react-datepicker-wrapper",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "react-datepicker__input-container",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "",
                    "name": "startDate",
                    "type": "text",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "name": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "name": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "react-datepicker-wrapper",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "react-datepicker__input-container",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "",
                  "name": "startDate",
                  "type": "text",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Select Your Start Date: ",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "label",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-testid": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-testid": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
